cmd/deploy/main.go

```diff
@@ -0,0 +1,71 @@
+package main
+
+import (
+	meteringclientv1 "github.com/operator-framework/operator-metering/pkg/generated/clientset/versioned/typed/metering/v1"
+	"github.com/operator-framework/operator-metering/pkg/operator/deploy"
+	apiextclientv1beta1 "k8s.io/apiextensions-apiserver/pkg/client/clientset/clientset/typed/apiextensions/v1beta1"
+
+	log "github.com/sirupsen/logrus"
+
+	"k8s.io/client-go/kubernetes"
+	_ "k8s.io/client-go/plugin/pkg/client/auth"
+	"k8s.io/client-go/tools/clientcmd"
+)
+
+var (
+	deployObj deploy.Deploy
+	logLevel  string
+)
+
+func main() {
+	logLevel = "info"
+	log.SetFormatter(&log.TextFormatter{
+		FullTimestamp:   true,
+		TimestampFormat: "01-02-2006 15:04:05",
+	})
+
+	logger := log.WithFields(log.Fields{
+		"app": "deploy",
+	})
+	logLevel, err := log.ParseLevel(logLevel)
+	if err != nil {
+		logger.WithError(err).Fatalf("invalid log level: %s", logLevel)
+	}
+	logger.Infof("Setting the log level to %s", logLevel.String())
+	logger.Logger.Level = logLevel
+
+	kubeconfig := clientcmd.NewNonInteractiveDeferredLoadingClientConfig(
+		clientcmd.NewDefaultClientConfigLoadingRules(),
+		&clientcmd.ConfigOverrides{},
+	)
+
+	restconfig, err := kubeconfig.ClientConfig()
+	if err != nil {
+		logger.Fatalf("Failed to initialize the kubernetes client config: %v", err)
+	}
+
+	client, err := kubernetes.NewForConfig(restconfig)
+	if err != nil {
+		logger.Fatalf("Failed to initialize the kubernetes clientset: %v", err)
+	}
+
+	apiextClient, err := apiextclientv1beta1.NewForConfig(restconfig)
+	if err != nil {
+		logger.Fatalf("Failed to initialize the apiextensions clientset: %v", err)
+	}
+
+	meteringClient, err := meteringclientv1.NewForConfig(restconfig)
+	if err != nil {
+		logger.Fatalf("Failed to initialize the metering clientset: %v", err)
+	}
+
+	deployObj.Client = client
+	deployObj.MeteringClient = meteringClient
+	deployObj.APIExtClient = apiextClient
+	deployObj.Logger = logger
+
+	err = deployObj.Driver()
```

Comment:
I think we should have a constructor which creates a `Deploy` object, and takes in the following as arguments:

```
	Logger                 log.FieldLogger
	Client                 *kubernetes.Clientset
	APIExtClient           apiextclientv1beta1.CustomResourceDefinitionsGetter
	MeteringClient         *metering.MeteringV1Client
```

The constructor could then call the `initDeployStructFields` and `initMeteringCRDSlice` methods before returning the `Deploy` struct. 

Then, we can call `Install` or `Uninstall` on the `deployObj` returned.